virtual_rainforest/core/logger.py

```diff
@@ -40,41 +40,48 @@
 so that ``DEBUG`` messages are suppressed, except when we are actively trying to debug
 the model.
 
-The logging module also defines a function that we should generally make use of to kill
-the simulation when something goes wrong. This function
-:func:`~virtual_rainforest.core.logger.log_and_raise` raises an exception (of the
-developers choice) and adds a developer specified ``CRITICAL`` message to the log. This
-function is useful as it ensures that ``CRITICAL`` logging events are accompanied by a
-simulation ending exception. As such, this is probably the only means by which you
-should log a ``CRITICAL`` message.
+Logging and exceptions
+----------------------
+
+When an exception is allowed to halt the code, it is important for the reason to be
+written to the log, as well as producing any traceback to the console. So, exception
+handling should always include a LOGGER call, using one of the following patterns.
+
+#. A test in the code indicates that we should raise an exception:
+
+  .. code-block:: python
+
+    if thing_has_gone_wrong:
+        to_raise = ValueError("It went wrong!")
+        LOGGER.critical(to_raise)
+        raise to_raise
+
+#. A ``try`` block results in an exception:
+
+  .. code-block:: python
+
+    try:
+        doing_something_that_raises()
+    except ValueError as excep:
+        LOGGER.critical(excep)
+        raise
+
+#. A ``try`` block results in an exception and we want to change the exception type:
```

Comment:
The below example doesn't actually change the exception type

Reply:
Huh. Good point!